sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java

```diff
@@ -0,0 +1,363 @@
+
+    private static final long serialVersionUID = 1L;
+    public String diversionAddress;
+    
+	/**
+     * Default constructor.
+     */
+    public Diversion() {
+        super(NAME);
+    }
+    
+    /** Default constructor given an address.
+    *
+    *@param address -- address of this header.
+    *
+    */
+
+   public Diversion(AddressImpl address) {
+       super(NAME);
+       this.address = address;
+   }
+   
+   public String encodeBody() {
+       return encodeBody(new StringBuilder()).toString();
+   }
+   /**
+    * Set the address member
+    *
+    * @param address Address to set
+    */
+   public void setDiversion(javax.sip.address.Address address) {
+       // Canonical form must have <> around the address.
+       if (address == null)
+           throw new NullPointerException("null address");
+       this.address = (AddressImpl) address;
+   }
+   
+   public void setDiversionAddress(String devAd) {
+	   diversionAddress = devAd;
+   }
+   
+   public Address getDiversion() {
+	   return address;
+   }
+   
+   public String getDiversionAddress() {
+	   return diversionAddress;
+   }
+   
+   /**
+    * Gets the Reason parameter of the Diversion header.
+    *
+    * @return the remote tag value
+    */
+   public String getReason() {
+   	 if (parameters == null)
+            return null;
+        return getParameter(ParameterNames.REASON);
+   }
+   /**
+    * Sets the remote tag of the Target-Dialog header.
+    *
+    * @param remoteTag the remote tag value to set
+    * @throws ParseException if the provided remoteTag is invalid
+    */
+   public void setReason(String r) throws ParseException {
+   	  if (r == null)
+             throw new NullPointerException("null param");
+         else if (r.trim().equals(""))
+             throw new ParseException("bad param", 0);
+         this.setParameter(ParameterNames.REASON, r);
+   }
+   /** Boolean function
+    * @return true if the parameter exist
+    */
+   public boolean hasReason() {
+       return hasParameter(ParameterNames.REASON);
+   }
+
+   /** remove Tag member
+    */
+   public void removeReason() {
+       parameters.delete(ParameterNames.REASON);      
+   }
+   
+   /**
+    * Gets the diversion-limit parameter from the address param list.
+    * @return field
+    */
+   public String getLimit() {
+       if (parameters == null)
+           return null;
+       return getParameter(ParameterNames.LIMIT);
+   }
+   
+   /**
+    * Sets the diversion-limit parameter of the Diversion header.
+    *
+    * @param reason diversion-limit parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setLimit(String l) throws ParseException {
+   	  if (l == null)
+             throw new NullPointerException("no limit param");
+         else if (l.trim().equals(""))
+             throw new ParseException("bad parameter", 0);
+         this.setParameter(ParameterNames.LIMIT, l);
```

Comment:
We can add a check to ensure it match the syntax of 2DIGIT at https://datatracker.ietf.org/doc/html/rfc5806#section-4 i.e: diversion-limit = "limit" "=" 1*2DIGIT

Reply:
Done